packages/components/tests/integration/components/hds/form/text-input/field-test.js

```diff
@@ -0,0 +1,115 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, resetOnerror } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | hds/form/text-input/field', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.afterEach(() => {
+    resetOnerror();
+  });
+
+  test('it renders the component', async function (assert) {
+    await render(hbs`<Hds::Form::TextInput::Field />`);
+    assert.dom('input').exists();
+  });
+  test('it should render the input with a specific CSS class', async function (assert) {
+    await render(hbs`<Hds::Form::TextInput::Field />`);
+    assert.dom('input').hasClass('hds-form-field__control');
+  });
+
+  // TYPE
+
+  test('it should render the "text" type if no type is declared', async function (assert) {
+    await render(hbs`<Hds::Form::TextInput::Field />`);
+    assert.dom('input').hasAttribute('type', 'text');
+  });
+  test('it should render the correct type depending on the @type prop', async function (assert) {
```

Comment:
[question] in the base tests we test for invalid `@type` values but don't here, is that intentional?

Reply:
_sidenote: i don't have a good suggestion otherwise but it strikes me how we are effectively copying the exact same tests from the base file into this wrapper component's tests which makes sense since we want to make sure consumers of the wrapper component have things work but feels also redudnant in a way I can't quite put my finger on_

Reply:
Yes I agree with you, but I went with redundancy in this case because we want to test that the property is correctly drilled down from the `field` to the `base` control.

Reply:
makes sense, but should we also test for the invalid types here as well or no?

Reply:
The invalid `@type` is tested in the `base` component (file `tests/integration/components/hds/form/text-input/base-test.js`, line 78), the `field` is consuming the `base` as inner component.